pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -1131,57 +1109,66 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	} else if c.clusterFlavor == cnstypes.CnsClusterFlavorWorkload {
 		// Check if it is WCP defined feature state.
 		if _, exists := common.WCPFeatureStates[featureName]; exists {
-			log.Infof("Feature %q is a WCP defined feature state. Reading the %q configmap in %q namespace.",
-				featureName, common.WCPCapabilityConfigMapName, common.KubeSystemNamespace)
-			// Check the `wcp-cluster-capabilities` configmap in supervisor for the FSS value.
-			if wcpCapabilityFssMap == nil {
-				wcpCapabilityConfigMap, err := c.k8sClient.CoreV1().ConfigMaps(common.KubeSystemNamespace).Get(ctx,
-					common.WCPCapabilityConfigMapName, metav1.GetOptions{})
+			log.Infof("Feature %q is a WCP defined feature state. Reading the capabilities CR %q.",
```

Comment:
Is there a write up on how we expect the supervisor capability to be used in CSI, when should a feature be considered fully enabled, how would CNS-CSI discover it, how that gets bubbled up to the guest cluster, when should CNS-CSI or PVCSI introduce its own feature states, etc?

Reply:
I have created a confluence page for this, will share the link with you.